### ISSUE TITLE
More feedback when configuration file is incorrectly formatted

### ISSUE THREAD

**Issue by Jaap-van-Hengstum** [twpayne/chezmoi#2695]
## Is your feature request related to a problem? Please describe.

I had an incorrectly formatted `chezmoi.toml` file and it took me quite a while to figure out why the configuration file wasn't read correctly. Running chezmoi normally didn't produce any warnings/errors and neither did the `dump-config` or `doctor` commands. It was only after I while that I discovered the `edit-config` command was giving me a warning for the incorrect configuration file.

Edit:
For example, when you forget the quotes in the toml file:

```
[edit]
    command = nvim
```

then `chezmoi doctor` still gives an ok result:
```
ok        config-file          ~/.config/chezmoi/chezmoi.toml, last modified 2023-01-15T21:40:19+01:00
```

## Describe the solution you'd like

1. `chezmoi doctor` should give a warning when the configuration file is incorrectly formatted
2. Chezmoi should always produce a warning when it tries to parse an incorrectly formatted configuration file, not only on certain commands.

**Comment by twpayne** [comment]
> 1. `chezmoi doctor` should give a warning when the configuration file is incorrectly formatted

Thanks, implementation in #2696.

> 2\. Chezmoi should always produce a warning when it tries to parse an incorrectly formatted configuration file, not only on certain commands.

I tried to reproduce this with `~/.config/chezmoi/chezmoi.toml` containing

```toml
[example]
    string = unquoted
```

When I ran `chezmoi status` I got the error message:

```
$ chezmoi status
chezmoi: invalid config: $WORK/home/user/.config/chezmoi/chezmoi.toml: $WORK/home/user/.config/chezmoi/chezmoi.toml: toml: incomplete number
```

Which commands currently don't produce a warning, but should?

**Comment by twpayne** [comment]
> Which commands currently don't produce a warning, but should?

On further investigation, `chezmoi dump-config` is one of these commands. Fixed in #2696.

This list of commands that run with invalid configuration files can be found with [this GitHub search](https://github.com/search?q=repo%3Atwpayne%2Fchezmoi%20doesNotRequireValidConfig&type=code).
